### PR TITLE
ukrainian translation (places)

### PR DESCRIPTION
**Pull request by burrbull**
~~See also #5380 ~~

Other in https://github.com/burrbull/fheroes2/tree/uk3

### CODE REVIEW

**Comment by zenseii** [comment]
> See also #5380

Hi @burrbull ! Thank you so much for your huge contribution! We usually have a rule stating that PRs of translations should not exceed 1000 additions of lines of code, and since this is already at 1300, we have to leave this PR at this point for now and wait for someone proficient in Ukrainian to review your translations.

We have set this limit due to earlier work on translations where we eventually had to split a PR because it was too big and this was slowing down the work on it, mostly due to how github works.

**Comment by zenseii** [comment]
Hi @burrbull ! When you have the time, could you please take a look at solving the merge conflicts now that the other Ukrainian PR has been merged? This is to be able to move on with the review of this PR. Thank you in advance!

**Comment by kant2002** [comment]
Also I would like if you can split your work on couple PRs so I can summon @zeule and @shprotru to help with reviewing. It's hard to review this large piece of work without missing something, because we are not professional translators.

Also I would suggest mark all translations as fuzzy for initial review. That way once all translation would be done, we can re-read again and catch small inconsistencies. That makes translation more polished and consistent in my experience.

File: files/lang/uk.po [hunk 1]
```diff
@@ -1978,16 +2021,21 @@ msgid ""
...
 
+#, fuzzy
 msgid "The Dungeon increases the income of the town by %{count} / day."
-msgstr ""
+msgstr "Підземелля підвищує прибуток міста на %{count} в день."
 
+#, fuzzy
 msgid "The Waterfall increases production of Centaurs by %{count} per week."
-msgstr ""
+msgstr "Підземелля підвищує приріст Кентаврів на %{count} за тиждень."
```
Comment (shprotru) [inline files/lang/uk.po:2030]: Водоспад

File: files/lang/uk.po [hunk 2]
```diff
@@ -2049,96 +2106,130 @@ msgstr "Гільдія Магів, 4 рівень"
...
 "day."
-msgstr ""
+msgstr "Замок покращує захист міста та підвищує прибуток на %{count} в день."
 
+#, fuzzy
 msgid ""
 "The Tent provides workers to build a castle, provided the materials and the "
 "gold are available."
-msgstr ""
+msgstr "Намет надає будівельників для замка."
```
Comment (shprotru) [inline files/lang/uk.po:2177]: ... за умовою вистачання матеріалів та золота для будівництва.

File: files/lang/uk.po [hunk 3]
```diff
@@ -2148,20 +2239,21 @@ msgid ""
 "moving the hero to the garrison."
 msgstr ""
 
+#, fuzzy
 msgid "This town may not be upgraded to a castle."
-msgstr "Розбудова замку в цьому мiстi неможлива."
+msgstr "Це місто неможливо покращити до замка."
```
Comment (shprotru) [inline files/lang/uk.po:2244]: Якась азіровщина з цими "покращеннями", попередній варіант, як на мене, краще.

File: files/lang/uk.po [hunk 4]
```diff
@@ -5113,9 +5211,8 @@ msgstr ""
 msgid "Show next hero"
 msgstr ""
 
-#, fuzzy
 msgid "Blood Morale"
-msgstr "Мораль"
+msgstr ""
```
Comment (shprotru) [inline files/lang/uk.po:5215]: Крові

File: files/lang/uk.po [hunk 5]
```diff
@@ -5886,29 +5984,36 @@ msgstr ""
...
 
+#, fuzzy
 msgid "Daemon Cave"
-msgstr ""
+msgstr "Печера Демона"
 
+#, fuzzy
 msgid "Faerie Ring"
-msgstr ""
+msgstr "Коло фей"
```
Comment (shprotru) [inline files/lang/uk.po:6001]: фей з великої

File: files/lang/uk.po [hunk 6]
```diff
@@ -5886,29 +5984,36 @@ msgstr ""
...
 
+#, fuzzy
 msgid "No object"
-msgstr ""
+msgstr "Пусте місце"
 
+#, fuzzy
 msgid "Alchemist Lab"
-msgstr ""
+msgstr "Алхімічна лабораторія"
```
Comment (shprotru) [inline files/lang/uk.po:5993]: лабораторія з великої?

File: files/lang/uk.po [hunk 7]
```diff
@@ -5963,716 +6078,952 @@ msgstr "Руїни"
 msgid "Fort"
 msgstr "Форт"
 
+#, fuzzy
 msgid "Abandoned Mine"
-msgstr ""
+msgstr "Покинута шахта"
```
Comment (shprotru) [inline files/lang/uk.po:6083]: в цьому контексті мені більш довподоби слово "штольня"

File: files/lang/uk.po [hunk 8]
```diff
@@ -5212,7 +5213,7 @@ msgid "Show next hero"
 msgstr ""
 
 msgid "Blood Morale"
-msgstr ""
+msgstr "Мораль Крові"
```
Comment (shprotru) [inline files/lang/uk.po:5216]: просто `Крові`

**Review by shprotru — APPROVED** [review]

**Comment by burrbull** [comment]
rebased on master

**Comment by ihhub** [comment]
@burrbull , thank you so much for taking care of this translation!

**Pull request merged**